### PR TITLE
support only bgen12

### PR DESCRIPTION
**Pull request by cseed**
Ported ImportBgenSuite to Python.  I mostly did this to make this simpler.
Removed some unused funtions in RichMatrixTable.

We can double-check with the community before merging.

### CODE REVIEW

**Comment by cseed** [comment]
Ah, I also nuked testBgenImportRandom with no replacement.  Is there any reason to think Gavin's example is not a complete test?

**Comment by jigold** [comment]
Looking at the corresponding `GEN` file, it looks like there's one missing value (first variant, first sample) and the rest of the values are close to 1. I think we're better off creating a new test dataset with random data. You can do this with `balding_nichols.export_gen` and then import/export with qctool.

File: python/hail/methods/impex.py [hunk 1]
```diff
@@ -818,11 +816,12 @@ def import_bgen(path,
...
     -----
 
-    Hail supports importing data from v1.1 and v1.2 of the
-    `BGEN file format <http://www.well.ox.ac.uk/~gav/bgen_format/bgen_format.html>`__.
-    For v1.2, genotypes must be **unphased** and **diploid**, and genotype
-    probability blocks must be compressed with zlib or uncompressed. If
-    `entry_fields` includes ``'dosage'``, all variants must be bi-allelic.
+    Hail supports importing data from v1.2 of the `BGEN file format
+    <http://www.well.ox.ac.uk/~gav/bgen_format/bgen_format.html>`__.
+    Genotypes **unphased** and **diploid**, genotype probabilities
```
Comment (jigold) [inline python/hail/methods/impex.py]: Genotypes must be ...

File: python/hail/tests/test_io.py [hunk 2]
```diff
@@ -75,3 +75,78 @@ def test_call_fields(self):
         entries = entries.key_by('locus', 's')
         entries = entries.select('GT', 'GTA', 'GTZ')
         self.assertTrue(entries._same(expected))
+        
+class BGENTests(unittest.TestCase):
```
Comment (jigold) [inline python/hail/tests/test_io.py]: I have a PR  (#3691) that refactors these tests as well. Please don't fix the other import types: VCF, PLINK.

**Comment by cseed** [comment]
There were some somewhat complicated merge conflicts in the tests, so if you could keep an eye out for mistakes there (e.g. dropped tests) that'd be awesome.

File: python/hail/tests/test_io.py [hunk 3]
```diff
@@ -452,6 +458,51 @@ def test_import_bgen_skip_invalid_loci(self):
...
+
+        sample_file = resource('example.sample')
+        bgen_file = resource('example.8bits.bgen')
+        hl.index_bgen(bgen_file)
+
+        bgenmt = hl.import_bgen(bgen_file, ['GP', 'dosage'], sample_file,
+                                contig_recoding=recoding)
+        et = bgenmt.entries()
+        et = et.transmute(gp_dosage = hl.gp_dosage(et.GP))
+        self.assertTrue(et.all(
```
[the remark this answers is not in the document]
Comment (jigold) [reply python/hail/tests/test_io.py]: I changed my mind on needing the additional checks. Approving as is.

**Comment by jigold** [comment]
I'm approving this PR, but I think we should add a post that tells users with BGEN 1.1 data to use Hail 0.1 or consider using QCTOOL to import/export to 1.2 format if they want to keep using Hail 0.2.

**Pull request merged**